### PR TITLE
feat(Menu): `<Menu>` Components GM-252

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

This adds a set of components to create structured and accessible ul / li menus for many use cases.  To avoid specificity issues and create certain amount of "Magic" the styles / configurations of descendants are managed through a shared context.  The variant and the spacing of the parent Menu is inherited by **all** descendants and cannot be misconfigured.
These work with color modes by default.

```tsx
<Menu variant="select">
  <MenuItem>Woah!</MenuItem>
  <MenuItem>Dude?</MenuItem>
  <MenuSeparator />
  <MenuItem active>Some Link</MenuItem>
</Menu>
```

Nested menus:
```tsx
<Menu variant="action">
    <MenuItem>
      <Menu>
        <MenuItem active>Some Link</MenuItem> 
      </Menu>
   </MenuItem>
</Menu>
```

**Components**
- `Menu` - A structured `ul` with a menu role the provisions a context for all descendents to read from.  All descendants inherit the style variants from the top level menu and have access to their relative depth.
- `MenuItem` - A `menuitem` component that renders `li` elements with appropriate roles.  Links are rendered as `li`'s with a role of none with an anchor as a child.
- `MenuSeparator` - a `li` with a role of separator which will only render within the `action` variant

**Variants**
- Select - For selection menus with background hover effects and tree structure.
- Navigation - A menu for site navigation that is persistent and has stark active states.
- Action - Used in dropdown menus that have menuitems with mixed behaviors (both links and selection options).

**Deletions**
- `Sidemenu`

#### Screenshots
Screen Shot 2021-07-15 at 11 29 00 AM
Screen Shot 2021-07-15 at 11 29 57 AM

### PR Checklist

- [x] Related to designs:
- [x] Related to JIRA ticket: https://codecademy.atlassian.net/browse/GM-252
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.23-alpha.e68dfb.0<br/>@codecademy/gamut-labs@19.5.1-alpha.e68dfb.0<br/>@codecademy/gamut@34.6.1-alpha.e68dfb.0<br/>@codecademy/styleguide@41.7.7-alpha.e68dfb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60c813b77ee058637ae53b01--gamut-preview.netlify.app>https://60c813b77ee058637ae53b01--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60c813b77ee058637ae53b01>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.23-alpha.58ce61.0<br/>@codecademy/gamut-labs@19.5.1-alpha.58ce61.0<br/>@codecademy/gamut@34.6.1-alpha.58ce61.0<br/>@codecademy/styleguide@41.7.7-alpha.58ce61.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60c8ab26bc836300b194111e--gamut-preview.netlify.app>https://60c8ab26bc836300b194111e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60c8ab26bc836300b194111e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.23-alpha.e44288.0<br/>@codecademy/gamut-labs@19.5.1-alpha.e44288.0<br/>@codecademy/gamut@34.6.1-alpha.e44288.0<br/>@codecademy/styleguide@41.7.7-alpha.e44288.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60c8ad347ee058ff6ee4a627--gamut-preview.netlify.app>https://60c8ad347ee058ff6ee4a627--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60c8ad347ee058ff6ee4a627>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.46-alpha.4c6ab9.0<br/>@codecademy/gamut-labs@19.11.2-alpha.4c6ab9.0<br/>@codecademy/gamut@34.10.9-alpha.4c6ab9.0<br/>@codecademy/styleguide@41.10.16-alpha.4c6ab9.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.49-alpha.f04409.0<br/>@codecademy/gamut-labs@19.11.5-alpha.f04409.0<br/>@codecademy/gamut@34.10.12-alpha.f04409.0<br/>@codecademy/styleguide@41.10.19-alpha.f04409.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.49-alpha.7bc88d.0<br/>@codecademy/gamut-labs@19.11.5-alpha.7bc88d.0<br/>@codecademy/gamut@34.10.12-alpha.7bc88d.0<br/>@codecademy/styleguide@41.10.19-alpha.7bc88d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60d605e14946a46ad745fb9b--gamut-preview.netlify.app>https://60d605e14946a46ad745fb9b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60d605e14946a46ad745fb9b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.49-alpha.0b480a.0<br/>@codecademy/gamut-labs@19.11.5-alpha.0b480a.0<br/>@codecademy/gamut@34.10.12-alpha.0b480a.0<br/>@codecademy/styleguide@41.10.19-alpha.0b480a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60d630216e159d066853a4d2--gamut-preview.netlify.app>https://60d630216e159d066853a4d2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60d630216e159d066853a4d2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.52-alpha.b5c69d.0<br/>@codecademy/gamut-labs@19.12.2-alpha.b5c69d.0<br/>@codecademy/gamut@34.11.1-alpha.b5c69d.0<br/>@codecademy/styleguide@41.10.22-alpha.b5c69d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60d6493543e8951f3396aae9--gamut-preview.netlify.app>https://60d6493543e8951f3396aae9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60d6493543e8951f3396aae9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.55-alpha.f326ff.0<br/>@codecademy/gamut-labs@19.13.1-alpha.f326ff.0<br/>@codecademy/gamut@34.12.1-alpha.f326ff.0<br/>@codecademy/styleguide@41.10.25-alpha.f326ff.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60da34ca0f910d008bdb5f38--gamut-preview.netlify.app>https://60da34ca0f910d008bdb5f38--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60da34ca0f910d008bdb5f38>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.58-alpha.d583c1.0<br/>@codecademy/gamut-labs@19.15.1-alpha.d583c1.0<br/>@codecademy/gamut@34.13.1-alpha.d583c1.0<br/>@codecademy/styleguide@41.12.1-alpha.d583c1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60db259392f4fad7cb04956f--gamut-preview.netlify.app>https://60db259392f4fad7cb04956f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60db259392f4fad7cb04956f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.61-alpha.afa3d8.0<br/>@codecademy/gamut-labs@19.15.4-alpha.afa3d8.0<br/>@codecademy/gamut@34.13.3-alpha.afa3d8.0<br/>@codecademy/styleguide@41.12.4-alpha.afa3d8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60de91b617a2764cd1288658--gamut-preview.netlify.app>https://60de91b617a2764cd1288658--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60de91b617a2764cd1288658>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.62-alpha.47de45.0<br/>@codecademy/gamut-labs@19.15.5-alpha.47de45.0<br/>@codecademy/gamut@34.14.1-alpha.47de45.0<br/>@codecademy/styleguide@41.13.1-alpha.47de45.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.63-alpha.7ac9c5.0<br/>@codecademy/gamut-labs@19.16.1-alpha.7ac9c5.0<br/>@codecademy/gamut@34.14.2-alpha.7ac9c5.0<br/>@codecademy/styleguide@41.14.1-alpha.7ac9c5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60edde7e3c963541da0f0d38--gamut-preview.netlify.app>https://60edde7e3c963541da0f0d38--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60edde7e3c963541da0f0d38>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.63-alpha.183420.0<br/>@codecademy/gamut-labs@19.16.1-alpha.183420.0<br/>@codecademy/gamut@34.14.2-alpha.183420.0<br/>@codecademy/styleguide@41.14.1-alpha.183420.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ef083bedad956162b91c5f--gamut-preview.netlify.app>https://60ef083bedad956162b91c5f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ef083bedad956162b91c5f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.63-alpha.f83763.0<br/>@codecademy/gamut-labs@19.16.1-alpha.f83763.0<br/>@codecademy/gamut@34.14.2-alpha.f83763.0<br/>@codecademy/styleguide@41.14.1-alpha.f83763.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f05129df94bd00950f6455--gamut-preview.netlify.app>https://60f05129df94bd00950f6455--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f05129df94bd00950f6455>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
🙌

File: packages/gamut/src/Menu/elements.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import {
+  styledOptions,
```
[the remark this answers is not in the document]
Comment (codecaaron) [reply packages/gamut/src/Menu/elements.tsx:2]: Not yet but its pretty limited in its uses outside of some our primitive components to ensure that prop typings don't conflict / get forwarded.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.63-alpha.b540c2.0<br/>@codecademy/gamut-labs@19.16.1-alpha.b540c2.0<br/>@codecademy/gamut@34.14.2-alpha.b540c2.0<br/>@codecademy/styleguide@41.14.1-alpha.b540c2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f08943836d5600a6cb23b8--gamut-preview.netlify.app>https://60f08943836d5600a6cb23b8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f08943836d5600a6cb23b8>Deploy Logs</a></p>

File: packages/gamut/src/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const activeProp = activePropnames[variant];
+  const computed = {
+    ...props,
+    ...rest,
+    variant: variant === 'select' ? 'select' : 'link',
+    role: 'menuitem',
+    [activeProp]: active,
+  } as ListItemProps;
+
+  if (href) {
```
Comment (jakemhiller) [inline packages/gamut/src/Menu/MenuItem.tsx:31]: noticed in the preview env that when you set the href property some of the styling seems to get messed up:

Screen Shot 2021-07-19 at 4 41 44 PM

**Review by jakemhiller — APPROVED** [review]
the one `href` prop styling issue aside this looks great!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.63-alpha.6440fa.0<br/>@codecademy/gamut-labs@19.16.1-alpha.6440fa.0<br/>@codecademy/gamut@34.14.2-alpha.6440fa.0<br/>@codecademy/styleguide@41.14.1-alpha.6440fa.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.63-alpha.aaf9a8.0<br/>@codecademy/gamut-labs@19.16.1-alpha.aaf9a8.0<br/>@codecademy/gamut@34.14.2-alpha.aaf9a8.0<br/>@codecademy/styleguide@41.14.1-alpha.aaf9a8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f5f89e2ffa5f37022a0ff9--gamut-preview.netlify.app>https://60f5f89e2ffa5f37022a0ff9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f5f89e2ffa5f37022a0ff9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.69-alpha.0eff38.0<br/>@codecademy/gamut-labs@19.16.7-alpha.0eff38.0<br/>@codecademy/gamut@35.0.2-alpha.0eff38.0<br/>@codecademy/styleguide@42.0.2-alpha.0eff38.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f5fcc188b15837664e6b5e--gamut-preview.netlify.app>https://60f5fcc188b15837664e6b5e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f5fcc188b15837664e6b5e>Deploy Logs</a></p>

**Pull request merged**